src/globus_sdk/services/flows/client.py

```diff
@@ -34,6 +34,137 @@ class FlowsClient(client.BaseClient):
+        :param input_schema: A JSON Schema to which Flow Invocation input must conform
+        :type input_schema: dict
+        :param subtitle: A concise summary of the flow’s purpose.
+        :type subtitle: str (0 - 128 characters), optional
+        :param description: A detailed description of the flow's purpose for end user
+            display.
+        :type description: str (0 - 4096 characters), optional
+        :param flow_viewers: A set of Principal URN values, or the value "public"
+            indicating entities who can view the flow
+
+            Examples:
+
+            .. code-block:: json
+
+                [ "public" ]
+
+            .. code-block:: json
+
+                [
+                    "urn:globus:auth:identity:b44bddda-d274-11e5-978a-9f15789a8150",
+                    "urn:globus:groups:id:c1dcd951-3f35-4ea3-9f28-a7cdeaf8b68f"
+                ]
+
+
+        :type flow_viewers: list[str], optional
+        :param flow_starters: A set of Principal URN values, or the value
+            "all_authenticated_users" indicating entities who can initiate a *Run* of
+            the flow
+
+            Examples:
+
+            .. code-block:: json
+
+                [ "all_authenticated_users" ]
+
+
+            .. code-block:: json
+
+                [
+                    "urn:globus:auth:identity:b44bddda-d274-11e5-978a-9f15789a8150",
+                    "urn:globus:groups:id:c1dcd951-3f35-4ea3-9f28-a7cdeaf8b68f"
+                ]
+
+        :type flow_starters: list[str], optional
+        :param flow_administrators: A set of Principal URN values indicating entities
+            who can perform administrative operations on the flow (create, delete,
+            update)
+
+            Example:
+
+            .. code-block:: json
+
+                [
+                    "urn:globus:auth:identity:b44bddda-d274-11e5-978a-9f15789a8150",
+                    "urn:globus:groups:id:c1dcd951-3f35-4ea3-9f28-a7cdeaf8b68f"
+                ]
+
+        :type flow_administrators: list[str], optional
+        :param keywords: A set of terms used to categorize the flow used in query and
+            discovery operations
+        :type keywords: list[str] (0 - 1024 items), optional
+        :param additional_fields: Additional Key/Value pairs sent to the create API
+        :type additional_fields: dict or str -> any, optional
+        """  # noqa E501
+
+        data = {
+            "title": title,
+            "definition": definition,
+            "input_schema": input_schema,
+            "subtitle": subtitle,
+            "description": description,
+            "flow_viewers": flow_viewers,
+            "flow_starters": flow_starters,
+            "flow_administrators": flow_administrators,
+            "keywords": keywords,
+            **(additional_fields or {}),
+        }
+
+        return self.post(
+            "/flows",
+            data={k: v for k, v in data.items() if v is not None},
```

Comment:
We had a discussion in the `run_flow` PR about a very similar pattern, and allowing `additional_fields` to dodge the automatic removal of `None` (since it's an intentional escape-valve). I'd like to see the same change here.

Since it's topical, I'll mention that we have an item for SDKv4 to add a new sentinel value (`MISSING`) so that we can disambiguate intentional-`None` vs omitted-parameters. That will apply a change to many methods in the SDK, including ones like these. That will likely make stripping `MISSING` out part of the `self.post()` call chain, which will let us more easily pass values "verbatim" in contexts like this one.
For now though, I just want us to treat `additional_fields` consistently across the existing methods.

Reply:
Sure, agreed.